src/native-common/View.tsx

```diff
@@ -82,7 +82,7 @@ export class View extends ViewBase<Types.ViewProps, {}> {
             if (handler) {
                 this._internalProps.allowDrop = true;
 
-                this._internalProps[name] = (e: React.SyntheticEvent) => {
+                this._internalProps[name] = (e: React.SyntheticEvent<View>) => {
```

Comment:
SyntheticEvent is typed in public typings.
It doesn't compile without this change.